### PR TITLE
Adding Sifive Coremark support

### PR DESCRIPTION
**Pull request by bsousi5**

### CODE REVIEW

**Comment by dconn-sifive** [comment]
This may not matter much but itim OVERFLOWS by 4046 bytes with the following build:

make PROGRAM=coremark TARGET=coreip-e21-arty CONFIGURATION=debug software

Perhaps we can document CONFIGURATION=release is recommended for coremark in general since no one should run -O0 when running a performance benchmark (even though coremark source override is -O3 in either case)

**Comment by bsousi5** [comment]
David,

Agree. We could also suggesting to modifying metal.ramrodata.lds to reduce ram size by 0x1000 and increase itim by 0x1000.

I’ll need to confirm with Alex but it does work

- Bunnaroath

File: doc/sphinx/contents.rst [hunk 1]
```diff
@@ -139,3 +139,6 @@ The example programs can be found under the ``software/`` directory.
 
   - "Dhrystone" Benchmark Program by Reinhold P. Weicker
 
+- coremark
+  - "CoreMark" Benchmark Program that measures the performance of embedded
+    microcrontrollers (MCU)
```
Comment (nategraff-sifive) [inline doc/sphinx/contents.rst]: This has some formatting issues.

File: scripts/standalone.mk [hunk 2]
```diff
@@ -37,7 +37,23 @@ ifeq ($(RISCV_CMODEL),)
...
+LINK_TARGET = ramrodata.coremark
+else
+LINK_TARGET = ramrodata
+endif
+
+ifneq (,$(filter $(TARGET),coreip-e76-rtl coreip-s76-rtl coreip-e76-arty coreip-s76-arty))
+RISCV_XCFLAGS += -O2 -fno-common -funroll-loops -finline-functions -funroll-all-loops --param max-inline-insns-auto=20 -falign-functions=8 -falign-jumps=8 -falign-loops=8 --param inline-min-speedup=10 -mtune=sifive-7-series -ffast-math
+else
+RISCV_XCFLAGS += -O2 -fno-common -funroll-loops -finline-functions --param max-inline-insns-auto=20 -falign-functions=4 -falign-jumps=4 -falign-loops=4 --param inline-min-speedup=10
+endif
```
Comment (nategraff-sifive) [inline scripts/standalone.mk]: These flags should go in the "Software Flags" section of this Makefile.

**Comment by nategraff-sifive** [comment]
Closing because I'm pretty sure #259 added this, feel free to re-open if I'm wrong.

**Pull request closed without merging**